packages/remix-dev/compiler.ts

```diff
@@ -442,6 +442,12 @@ function createServerBuild(
       format: config.serverModuleFormat,
       treeShaking: true,
       minify: options.mode === BuildMode.Production && isCloudflareRuntime,
+      // The type of dead code elimination we want to do depends on the
+      // minify syntax property: https://github.com/evanw/esbuild/issues/672#issuecomment-1029682369
+      // without this we have dev builds that run fine in production, but
+      // ship server code to the browser, blowing it up due to not optimizing
+      // out imports and module level blocks like `if (process.env.NODE_ENV === "test") {`.
```

Comment:
I took a stab at some alternate wording that I think flows better and makes more sense to me, but feel free to take or leave as much of this wording as you think is best.

```suggestion
      // For dev builds, we need dead code elimination (beyond what's provided by `treeShaking: true`) so that
      // we don't ship server code to the browser and eliminate non-dev blocks (e.g. `if (process.env.NODE_ENV === "test")`).
      // To do so, we enable the `minifySyntax` option, which "does more advanced dead code elimination during syntax compression".
      // See https://github.com/evanw/esbuild/issues/672#issuecomment-1029682369
```